src/main/java/anto/Parser.java

```diff
@@ -0,0 +1,206 @@
+package anto;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Parser class handles all reading of inputs from command line and decides what to do.
+ */
+public class Parser {
+
+    private Ui ui;
+    private TaskList taskList;
+    private Scanner scanner = new Scanner(System.in);
+
+    /**
+     * Creates a Parser class.
+     * @param ui Ui class that handles printing to command line.
+     * @param taskList TaskList with all the current tasks.
+     */
+    public Parser(Ui ui, TaskList taskList) {
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Continuously reads inputs from command line and decide what actions to take.
+     */
+    public void readInputs() {
+        this.ui.greet();
+
+        while (this.scanner.hasNext()) {
+            try {
+                String input = this.scanner.nextLine();
+
+                if (input.equals("bye")) {
+                    this.ui.sayBye();
+                    return;
+                } else if (input.equals("list")) {
+                    if (this.taskList.getLength() == 0) {
+                        this.ui.printNoTasks();
+                    } else {
+                        this.ui.printList();
+                    }
+                } else if (input.startsWith("mark")) {
+                    // Throw error if there is no index
+                    if (input.length() < 6 || input.split(" ", 2).length < 2) {
+                        throw new InvalidIndexException("OOPS!!! You need to include an index to mark task.");
+                    }
+
```

Comment:
Good that all your logical units within a block are separated by one blank line.